File: Night at the Museum.py
d = dict()
for i in range(97, 123):
    d[chr(i)] = abs(97-i)+1

n = list(input())
a = d['a']
r = 0
for i in n:
    b = d[i]
    r += min(abs(b-a), 26-abs(b-a))
    a = b
print(r)

File: Night at the Museum.cpp
#include<bits/stdc++.h>

using namespace std;

int main()
{
    map<char,int>d;
    int a = 1;
    for (char i = 'a' ; i <= 'z' ; ++i)
    {
        d.insert({i,a});
        a++;
    }

    string s;
    cin >> s;
    int x = d['a'],y,r=0;
    for(int i = 0; i < s.size() ; ++i)
    {
        y = d[s[i]];
        r += min(abs(y - x),26-abs(y-x));
        x = y;
    }
    cout << r << "\n";
    return 0;
}
